akka-stream-tests/src/test/scala/akka/stream/io/StreamConvertersAsInputStreamErrorSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.io
+
+import java.io.{IOException, InputStream}
+import java.util.concurrent.ThreadLocalRandom
+
+import akka.stream._
+import akka.stream.scaladsl.{Source, StreamConverters}
+import akka.stream.testkit.Utils._
+import akka.stream.testkit._
+import akka.util.ByteString
+import akka.NotUsed
+
+/**
+ * https://github.com/akka/akka/issues/25614
+ * Behavior does not change if you use Supervision.stoppingDecider or not, or
+ * dispatcher: akka.actor.default-dispatcher or akka.stream.default-blocking-io-dispatcher
+ *
+ * All combinations have been tested. Additionally, using
+  * val (probe, inputStream) = TestSource.probe[ByteString].toMat(StreamConverters.asInputStream())(Keep.both).run()
+ * fails too soon in the flow when attempting to on-demand raising of an exception with probe.send RuntimeException.
+ */
+abstract class StreamConvertersAsInputStreamErrorSpec extends StreamSpec(UnboundedMailboxConfig) {
+  val decider: Supervision.Decider = Supervision.stoppingDecider
+  implicit val mat = ActorMaterializer(ActorMaterializerSettings(system).withSupervisionStrategy(decider))
+
+  private def source: Source[ByteString, NotUsed] =
+    Source(List(0, 1, 2, 3, 4))
+      .map(i ⇒ if (i == 3) throw new RuntimeException("failed") else ByteString(i))
```

Comment:
The flows are in general immutable blueprints that can be re-used so this one can be a val and still be materialized several times and those running streams will not have anything in common with each other.